clients/src/main/java/org/apache/kafka/clients/consumer/internals/AsyncKafkaConsumer.java

```diff
@@ -398,6 +387,38 @@ public class AsyncKafkaConsumer<K, V> implements ConsumerDelegate<K, V> {
                 requestManagersSupplier);
     }
 
+    private Optional<ConsumerGroupMetadata> initializeGroupMetadata(final ConsumerConfig config,
+                                                                    final GroupRebalanceConfig groupRebalanceConfig) {
+        final Optional<ConsumerGroupMetadata> groupMetadata = initializeGroupMetadata(
+            groupRebalanceConfig.groupId,
+            groupRebalanceConfig.groupInstanceId
+        );
+        if (!groupMetadata.isPresent()) {
+            config.ignore(ConsumerConfig.AUTO_COMMIT_INTERVAL_MS_CONFIG);
+            config.ignore(THROW_ON_FETCH_STABLE_OFFSET_UNSUPPORTED);
+        }
+        return groupMetadata;
+    }
+
+    private Optional<ConsumerGroupMetadata> initializeGroupMetadata(final String groupId,
+                                                                    final Optional<String> groupInstanceId) {
+        if (groupId != null) {
+            if (groupId.isEmpty()) {
+                throwInInvalidGroupIdException();
+                return Optional.empty();
```

Comment:
Unreachable!

Reply:
Yes, I know, but without it won't compile because it needs a return statement in each branch. However, I think I found a better solution, now.